### PR TITLE
refactor: sort ops before building witness

### PR DESCRIPTION
**Pull request by rphmeier**
this is cleaner too

### CODE REVIEW

**Comment by rphmeier** [comment]
* **#67** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/67?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rphmeier and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: nomt/src/store.rs [hunk 1]
```diff
@@ -156,4 +156,10 @@ impl Transaction {
             Some(value) => self.batch.put_cf(&cf, page_id.to_bytes().as_ref(), value),
         }
     }
+
+    /// Write the root to metadata.
+    pub fn write_root(&mut self, root: Node) {
+        let cf = self.shared.db.cf_handle(METADATA_CF).unwrap();
+        self.batch.put_cf(&cf, b"root", &root[..]);
+    }
```
[the remark this answers is not in the document]
Comment (rphmeier) [reply nomt/src/store.rs:164]: yes, will unstage

[the remark this answers is not in the document]
Comment (rphmeier) [reply nomt/src/store.rs]: (#76 )

**Comment by pepyakin** [comment]
What happened to this one?

**Pull request closed without merging**